airbyte-ci/connectors/metadata_service/orchestrator/orchestrator/sensors/metadata.py

```diff
@@ -0,0 +1,45 @@
+from dagster import sensor, RunRequest, SkipReason, SensorDefinition, SensorEvaluationContext, build_resources, DefaultSensorStatus
+from orchestrator.utils.dagster_helpers import deserialize_composite_etag_cursor, serialize_composite_etag_cursor
+
+
+def metadata_updated_sensor(job, resources_def) -> SensorDefinition:
+    """
+    This sensor is responsible for polling the metadata folder in GCS for new or updated metadata files.
+    If it notices that the etags have changed, it will trigger the given job.
+    """
+
+    @sensor(
+        name=f"{job.name}_on_metadata_updated",
+        job=job,
+        minimum_interval_seconds=30,
+        default_status=DefaultSensorStatus.STOPPED,
+    )
+    def metadata_updated_sensor_definition(context: SensorEvaluationContext):
+        context.log.info("Starting gcs_metadata_updated_sensor")
+
+        with build_resources(resources_def) as resources:
+            context.log.info("Got resources for gcs_metadata_updated_sensor")
+
+            etag_cursor_raw = context.cursor or None
+            etag_cursor = deserialize_composite_etag_cursor(etag_cursor_raw)
```

Comment:
nit: the `etag_cursor` name made me think we were handling a single etag, but it's a list of etags.
`etags_cursor`?